core/src/main/java/org/apache/iceberg/UpdateRequirement.java

```diff
@@ -62,12 +62,31 @@ public void validate(TableMetadata base) {
             "Requirement failed: UUID does not match: expected %s != %s", base.uuid(), uuid);
       }
     }
+  }
+
+  class AssertViewUUID implements UpdateRequirement {
+    private final String uuid;
+
+    public AssertViewUUID(String uuid) {
+      Preconditions.checkArgument(uuid != null, "Invalid required UUID: null");
+      this.uuid = uuid;
+    }
+
+    public String uuid() {
+      return uuid;
+    }
+
+    @Override
+    public void validate(TableMetadata base) {
+      throw new ValidationException(
+          "Cannot validate %s against a table", this.getClass().getSimpleName());
```

Comment:
it's not. We only throw for `ViewMetadata` but not for `TableMetadata` in https://github.com/apache/iceberg/pull/8831/files#diff-3594848799883f9d669511640aadba2b625ce3e07b93a2ae211fcc2bb999e0aaR28, but we could change that. Let me know if you'd like me to make that change

Reply:
I went ahead and moved this to the base class, so that we're in line with `MetadataUpdate`: https://github.com/apache/iceberg/blob/d9eb937bfb6bfed98b0a7e64ef402792a1a2b777/core/src/main/java/org/apache/iceberg/MetadataUpdate.java#L31-L39